velox/vector/arrow/Bridge.cpp

```diff
@@ -101,6 +107,9 @@ class VeloxToArrowBridgeHolder {
   std::unique_ptr<ArrowArray*[]> children_;
 
   std::unique_ptr<ArrowArray> dictionary_;
+
+  // Keep track of used base vectors for dictionary processing.
```

Comment:
Can you explain why we cannot use `buffers_` above? Why do we need this additional vector?

Reply:
The idea was that we create a new vector with new memory and copy values.
When the vector goes out pf scope the BufferPtr of the holder points to the values buffer of the vector.
This isn't a problem with the passed in vector that comes from the higher layers.

I assumed I need to keep the vector around so the values are still legit by the time the BufferPtr is written to the file which is as long as the holder is not deleted.
But looks like they are allocated as shared_ptr and then moved to be a member. So looks like we might not need to hold on to the vector as one holder of the BufferPtr for values.